vcf2fhir/json_generator.py

```diff
@@ -7,19 +7,37 @@
 general_logger = logging.getLogger("vcf2fhir.general")
 
 
-def _valid_record(record):
+def _valid_record(record, genomic_source_class):
+    if record.is_sv and len(record.samples) > 1:
+        invalid_record_logger.debug(
+            ("Reason: Multi-sample VCFs for SVs are not allowed, " +
+             "Record: %s, considered sample: %s"),
+            record,
+            record.samples[0].data)
+        return False
     if not (validate_chrom_identifier(record.CHROM)):
         invalid_record_logger.debug(
             ("Reason: VCF CHROM is not recognized, " +
              "Record: %s, considered sample: %s"),
             record,
             record.samples[0].data)
         return False
-    if(record.is_sv):
+    if(
+            record.is_sv and
+            record.INFO['SVTYPE'] not in ['INS', 'DEL', 'DUP', 'CNV', 'INV']):
         invalid_record_logger.debug(
-            ("Reason: VCF INFO.SVTYPE is present. " +
-             "(Structural variants are excluded), " +
-             "Record: %s, considered sample: %s"),
+            ("Reason: VCF INFO.SVTYPE must be in ['INS', 'DEL', " +
+             "'DUP', 'CNV', 'INV']. Record: %s, considered sample: %s"),
+            record,
+            record.samples[0].data)
+        return False
+    if(
+            record.is_sv and
+            record.INFO['SVTYPE'] in ['CNV', 'DUP', 'DEL'] and
```

Comment:
> record.INFO['SVTYPE'] in ['CNV', 'DUP', 'DEL']

It's repeated multiple times, it will be a good idea to add a utility for it something like `is_<what you are checking>(record) `. However, I am not able to able to understand why this condition is required. can you point me to a location in our design document form where you pick this condition?

Reply:
Actually,  prior to to the changes in the design document, we were supposed to raise an exception if `CN` is not present when `SVTYPE` is in `['CNV', 'DUP', 'DEL']` but now we have to populate the `copyNumber` component if possible. I added this condition to ignore the **complete record** and add it to the `invalid record log` if the condition is `true`. I just want to confirm with @rhdolin if we have to ignore just the `copyNumber` component or the **complete record** If the condition evaluates to true.

Reply:
Just ignore the component, and still translate the record.